### DISCUSSION TITLE
Feature Request: Provide Lineage for DAX table variables in the Watch Window

### DISCUSSION THREAD

**Discussion by corycundy** [TabularEditor/TabularEditor3#772] [Ideas]
It is great to be able to add a variable that is a table to the Watch window and have the Generate query functionality. What would be extra nice is to have the ability to somehow see which model table that table variable is reading. 

I know I can find out by clicking Generate query and looking at the code, but it would be nice to see without having to do that extra step. 

One idea would be to add it to the tool tip when hovering over the Value or Type column for that expression.

**Comment by otykier** [comment]
Great suggestion, thanks!

**Comment by corycundy** [comment]
Your welcome! This would be especially handy when you have role playing dimensions in your model.

**Comment by corycundy** [comment]
After thinking about my request above, I think it would be even more helpful to somehow show the lineage of tables and columns from the watch window. 

For example, a table variable could exist within a measure that has columns from multiple tables. I can currently add the table variable to watch and see the high level information or open it up in a Watch Window tab. What would be nice is if there was an additional context menu item that said "Show Table Lineage" for table expressions and it would pop-up a dialog with each a row for each column in the table along with the table it has lineage to. It could pop up a dialog similar to how the row context iteration dialog works within the debugger. 

It would be great to have this in both the Locals Window and Watch Window.

Here is an example, of where the feature would be nice.

The dialog would show the follow table:

Column Name               |   Underlying Column Name
----------------------------------------------------------
Education                       |  'Customer'[Education]
Color                              |  'Product'[Color]
No Lineage                    |

DAX Query to Debug
<img width="540" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/64218283/85c5a7de-4466-4346-b4cd-a3fd7e9a710c">

Debugging the Measure
<img width="558" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/64218283/a82894d2-c279-4cfc-8d9e-1fdfce859145">

**Comment by mlonsk** [comment]
Hi @corycundy 
Thank you for providing the extra feedback. I think it is a great suggestion and we will keep this in mind when we next make an upgrade to the DAX debugger. I do not however have a timeline for that currenlty.
